openbb_platform/core/openbb_core/provider/utils/helpers.py

```diff
@@ -96,6 +96,254 @@ def get_querystring(items: dict, exclude: List[str]) -> str:
+    _session: requests.Session = kwargs.pop("session", None) or requests.Session()
+    _session.headers.update(headers)
+
+    if (
+        python_settings.get("cafile") is not None
+        and os.environ.get("REQUESTS_CA_BUNDLE") is not None
+    ):
+        _session.verify = (
+            combine_certificates(
+                python_settings["cafile"], os.environ.get("REQUESTS_CA_BUNDLE")
+            )
+            if python_settings["cafile"] != os.environ.get("REQUESTS_CA_BUNDLE")
+            else combine_certificates(python_settings["cafile"])
+        )
+    elif python_settings.get("cafile") is not None:
+        _session.verify = combine_certificates(python_settings["cafile"])
+    elif (
+        os.environ.get("REQUESTS_CA_BUNDLE") is not None
+        and python_settings.get("cafile") is None
+    ):
+        certs = os.environ.get("REQUESTS_CA_BUNDLE", "")
+        _session.verify = combine_certificates(certs)
+    elif python_settings.get("verify_ssl") is False:
+        _session.verify = False
+
+    if python_settings.get("certfile"):
+        _session.cert = (
+            (python_settings["certfile"], python_settings.get("keyfile", ""))  # type: ignore
+            if python_settings.get("keyfile")
+            else python_settings["certfile"]
+        )
+
+    if (
+        os.environ.get("HTTP_PROXY") or os.environ.get("HTTPS_PROXY")
+    ) and not python_settings.get("proxy"):
+        http = os.environ.get("HTTP_PROXY", os.environ.get("HTTPS_PROXY", ""))
+        https = os.environ.get("HTTPS_PROXY", os.environ.get("HTTP_PROXY", ""))
+        proxies: dict = {}
+        if http:
+            proxies["http"] = http
+        if https:
+            proxies["https"] = https
+        _session.proxies = proxies
+    elif python_settings.get("proxy"):
+        _session.proxies = {
+            "http": python_settings["proxy"],
+            "https": python_settings["proxy"],
+        }
+
+    if python_settings.get("cookies"):
+        _session.cookies = (
+            python_settings["cookies"]
+            if isinstance(
+                python_settings["cookies"], requests.cookies.RequestsCookieJar
+            )
+            else requests.cookies.cookiejar_from_dict(
+                python_settings.get("cookies", {})
+            )
+        )
+
+    if python_settings.get("auth"):
+        _session.auth = (
+            python_settings["auth"]
+            if isinstance(python_settings["auth"], (tuple, requests.auth.AuthBase))
+            else tuple(python_settings["auth"])
+        )
+
+    if kwargs:
+        for key, value in kwargs.items():
+            try:
+                if hasattr(_session, key):
+                    if hasattr(getattr(_session, key, None), "update"):
+                        getattr(_session, key, {}).update(value)
+                    else:
+                        setattr(_session, key, value)
+            except AttributeError:
+                continue
```

Comment:
I don't see this being used anywhere, what's the purpose of updating the keys of kwargs?

Reply:
In the sync/async functions the kwargs are applied in the function, but if you were to use this as a standalone function to create a session object, you might want to set something not defined by `python_settings`.

This was the intention at least.